crates/node-builder/src/exex.rs

```diff
@@ -11,7 +11,7 @@ use reth_tasks::TaskExecutor;
 use std::{borrow::Cow, pin::Pin};
 
 /// An ExEx (Execution Extension) that processes new blocks and emits events on a stream.
-pub trait ExEx: Stream<Item = ExExEvent> + Send + 'static {
+pub trait ExEx: Stream<Item = eyre::Result<ExExEvent>> + Send + 'static {
```

Comment:
after thinking about this, this should really be a future 

and we provide a sender for events when we create the exex

Reply:
so `launch(ctx, sender) -> impl Future<Output = Result<(), E>>` impld for `FnOnce(ctx, sender) -> ..`?

Reply:
yeah, depending on what additional info we need in the exex trait we can even do:

`fn install_exex(&self, name: String, FnOnce -> impl Future<Output= impl Future>`